File: app/src/main/java/fr/free/nrw/commons/di/CommonsDaggerIntentService.java
package fr.free.nrw.commons.di;

import android.app.IntentService;
import android.app.Service;
import dagger.android.AndroidInjector;

public abstract class CommonsDaggerIntentService extends IntentService {

  public CommonsDaggerIntentService(String name) {
    super(name);
  }

  @Override
  public void onCreate() {
    inject();
    super.onCreate();
  }

  private void inject() {
    ApplicationlessInjection injection = ApplicationlessInjection
        .getInstance(getApplicationContext());

    AndroidInjector<Service> serviceInjector = injection.serviceInjector();

    if (serviceInjector == null) {
      throw new NullPointerException("ApplicationlessInjection.serviceInjector() returned null");
    }

    serviceInjector.inject(this);
  }

}

File: app/src/main/java/fr/free/nrw/commons/di/CommonsDaggerIntentService.kt
package fr.free.nrw.commons.di

import android.app.IntentService

/**
 * Base class for intent services that previously used Dagger Android injection.
 *
 * NOTE: This class is DEPRECATED with Hilt. IntentServices should use WorkManager with Hilt Workers instead.
 * This class is kept as a simple IntentService extension for backward compatibility,
 * but all injection functionality has been removed.
 */
abstract class CommonsDaggerIntentService(name: String?) : IntentService(name)

